src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -322,9 +323,10 @@ protected virtual Parameter[] BuildMethodParameters(IReadOnlyDictionary<InputPar
                         // This method has a flattened body
                         if (bodyRequestParameter.Kind == InputOperationParameterKind.Flattened && library != null)
                         {
-                            var objectType = bodyRequestParameter.Type switch
+                            (InputType inputType, bool isNullable) = bodyRequestParameter.Type is InputNullableType nullableType ? (nullableType.Type, true) : (bodyRequestParameter.Type, false);
+                            var objectType = inputType switch
                             {
-                                InputModelType inputModelType => library.ResolveModel(inputModelType).Implementation as SerializableObjectType,
+                                InputModelType inputModelType => library.ResolveModel(inputModelType).WithNullable(isNullable).Implementation as SerializableObjectType,
```

Comment:
WithNullable here is actually redundant.
And if we care, we should do it through `TypeFactory.CreateType` which handles nullability automatically, you do not even need to unwrap the nullable type like we are doing above.

Reply:
No. This is static function, there is no `TypeFactory` instance. so we cannot use `TypeFactory.CreateType`. We may need to refactor this in future. maybe an static `TypeFactory` variable is needed.

Current we will keep this.

Reply:
TypeFactory is available from the public static method, we can pass it along to fix this

Reply:
It could use library because we pass the library along.
Here we do not need the library, instead we should change it to TypeFactory.

Reply:
So if we do not plan to do the typefactory thing (despite I think it just harms nothing and make things clear), please remove the useless `WithNullable` here.

Reply:
we could fix this later